### PR TITLE
Merging docs for ParseXML.py

### PR DESCRIPTION
**Pull request by mattagape**
Hi Sarah

I think this is now mostly done.

Please pay particular attention to comments annotated with "Sarah", "TODO", and "issue", although obviously it would be good if you checked the whole thing!

PS One last thing we could do (on all the codebase), but I would suggest is a low priority for now, is to get the code to comply with PEP-8. We can run `pycodestyle` and `flake8` on the code; I guess ideally they would form part of our CI process. we could always do this later.

### CODE REVIEW

File: generator/parseXML/ParseXML.py [hunk 1]
```diff
@@ -218,28 +220,70 @@ def get_bool_value(self, node, name):
...
+        Example <attribute> node:
+        <attribute name="sampledVolume" required="false" type="lo_element"
+         element="SampledVolume" abstract="false"/>
+        '''
         element = ''
-        temp = self.get_value(node, 'element')
+        temp = self.get_value(node, 'element')  # e.g. "SampledVolume"
         if temp is not None:
-            element = strFunctions.standard_element_name(temp)
+            element = strFunctions.standard_element_name(temp)  # q.v.
```
Comment (skeating) [inline generator/parseXML/ParseXML.py]: what is q.v. ?

File: generator/parseXML/ParseXML.py [hunk 2]
```diff
@@ -778,23 +980,70 @@ def get_elements_for_version(self, pkg_node):
...
+          <attributes>
+            <attribute name="x" required="true" type="double" abstract="false"/>
+            <attribute name="y" required="true" type="double" abstract="false"/>
+            <attribute name="point" required="false" ... element="Point" ... abstract="false"/>
+          </attributes>
+        </element>
+
+        the third <attribute> node is a "child" of the "Point" element.
+
+        TODO: Sarah, does the <attribute> node have to be nested within its parent <element>
```
Comment (skeating) [inline generator/parseXML/ParseXML.py]: Yes <attribute> will always be nested within <element> 
you are correct parent and child is synononymous with outer/inner
but we do have grandparents as well

**Review by skeating — APPROVED** [review]
couple of questions and made issues of some things

File: generator/parseXML/ParseXML.py (hunk 1 above)
Comment (mattagape) [reply generator/parseXML/ParseXML.py]: It means "which see". I meant, "look at function `strFunctions.standard_element_name()` for further details. but I guess this is rather a pointless comment, so I've removed it.

File: generator/parseXML/ParseXML.py (hunk 2 above)
Comment (mattagape) [reply generator/parseXML/ParseXML.py]: Thanks - I have updated the comment to resolve this.

**Review by skeating — APPROVED** [review]

**Pull request merged**